File: ztuo/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> temp = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            if(!temp.containsKey(target - nums[i])) {
                temp.put(nums[i], i);
            }
            else {
                res[0] = i; res[1] = temp.get(target - nums[i]);
                break;
            }
        }
        return res;
    }
}

File: syli/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> res;
        unordered_map<int, int> temp;
        
        for(int i=0; i<nums.size(); i++){
            int dif = target - nums[i];
            if(temp.find(dif) != temp.end()){
                res.push_back(temp[dif]);
                res.push_back(i);
                return res;
            }
            else temp[nums[i]] = i;
        }
    }
};
